### PR TITLE
Remove async paths from EOP constructor

### PR DESCRIPTION
**Pull request by ggetz**
Should fix https://github.com/CesiumGS/cesium/issues/10970

* Moves async functionality in the EOP constructor to `fromUrl` (this is a private API so no deprecation or `CHANGES.md` update necessary)
* Fixup error handling so there should only be one error thrown
* Fixup specs to reflect new error handling

I believe this is the first PR to use async/await. I figured it's time given we're about to submit more async changes as part of https://github.com/CesiumGS/cesium/issues/10909, and this should help flatten a code for better readability.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ggetz!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/no-async-eop-constructor/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: packages/engine/Source/Core/EarthOrientationParameters.js [hunk 1]
```diff
@@ -19,10 +20,6 @@ import TimeStandard from "./TimeStandard.js";
  * @constructor
```
Comment (sanjeetsuhag) [inline packages/engine/Source/Core/EarthOrientationParameters.js:20]: Should we add a warning here that this should typically not be constructed directly, like we do with `Model.js`?

https://github.com/CesiumGS/cesium/blob/b9106c0a915be1b59d1f35878ebbbe6f97505f43/packages/engine/Source/Scene/Model/Model.js#L42

**Review by sanjeetsuhag — COMMENTED** [review]
This is a great improvement for readability and feels like cleaner code overall! Love that we don't have to have smaller state/promise variables hanging around.

File: packages/engine/Source/Core/EarthOrientationParameters.js (hunk 1 above)
Comment (ggetz) [reply packages/engine/Source/Core/EarthOrientationParameters.js:21]: There are cases where the constructor may be used. I've add a disclaimer similar to `IonResource` - 

> This object is normally not instantiated directly, use {@link EarthOrientationParameters.fromUrl}.

**Comment by ggetz** [comment]
Thanks @sanjeetsuhag! Updated!

**Comment by sanjeetsuhag** [comment]
Thanks @ggetz!

**Pull request merged**